socorro/siglists/__init__.py

```diff
@@ -0,0 +1,54 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+
+
+_DIRECTORY = os.path.dirname(os.path.abspath(__file__))
+
+
+# This is a hack because sentinels can be a tuple, with the second item being
+# a function to verify if the sentinel applies. It's quite hard to express
+# that in a .txt file, so this special value is here. This list should not
+# grow much, and if it does, we should find a better solution for handling
+# these special values.
+_SPECIAL_EXTENDED_VALUES = {
+    'signature_sentinels': (
+        (
+            'mozilla::ipc::RPCChannel::Call(IPC::Message*, IPC::Message*)',
+            lambda x: (
+                'CrashReporter::CreatePairedMinidumps(void*, '
+                'unsigned long, nsAString_internal*, nsILocalFile**, '
+                'nsILocalFile**)'
+            ) in x
+        ),
+    ),
+}
+
+
+def _get_file_content(source):
+    """Return a tuple, each value being a line of the source file.
+
+    Remove empty lines and comments (lines starting with a '#').
+    """
+    filepath = os.path.join(_DIRECTORY, source + '.txt')
+
+    with open(filepath, 'r') as f:
+        lines = tuple(
+            line.replace('\n', '') for line in f
+            if line.strip() and not line.strip().startswith('#')
+        )
```

Comment:
I'd LOVE to see something like this instead:
```python
import re

with open(filepath) as f:
    for line_no, line in f:
        # ...check that it's not empty, stripped and not starting with #
        try:
            re.compile(line)
         except Exception as exception:
             raise BadRegularExpressionLineError(exception.message + ' (line %d in %s)' % (line_no + 1, filepath))
        lines += (line,)
```

That would make it a million times easier to spot what mistake has been made when people make a PR and look at Travis failures.